src/psyclone/psyir/tools/read_write_info.py

```diff
@@ -158,10 +165,47 @@ def add_write(self, signature, container_name=None):
         self._sorted = False
 
     # -------------------------------------------------------------------------
-    def is_read(self, signature):
-        ''':returns: whether the signature is in the read list (independent \
+    def is_read(self, signature: Signature) -> bool:
+        '''
+        Checks if the given signature is in the read list.
+
+        :param signature: the signature to remove
+
+        :returns: whether the signature is in the read list (independent
             of the container name).
-        :rtype: bool
 
         '''
         return any(signature == sig for _, sig in self._read_list)
+
+    # -------------------------------------------------------------------------
+    def remove(self, signature: Signature, container_name: str = "") -> None:
+        '''
+        This function removes a signature (if required with the corresponding
```

Comment:
Is `container_name` Optional? If so pls update the typehint and explain when container_name should be provided and what happens if it isn't.

Reply:
Indeed. Updated the typing and comments.